index.js

```diff
@@ -1,21 +1,15 @@
 var loaderUtils = require('loader-utils');
 var MessageFormat = require('messageformat');
 
-module.exports = function(content) {
-  var query = loaderUtils.parseQuery(this.query);
-  var locale = query.locale || 'en';
-  var messages = (Array.isArray(this.inputValue) && typeof this.inputValue[0] === 'object') ? this.inputValue[0] : this.exec(content, this.resource);
```

Comment:
Looks like you switched this to `JSON.parse(messages)` below. That won't actually work with `json-loader` or `multi-json-loader` which export JS (`module.exports = "<json string>"`). I know it's nice to be able to just write `require('messageformat-loader!./messages.json');`, but that's not quite flexible enough for my use case (combine a whole directory of JSON messages files into a single output object).

If you want, we could wrap the `JSON.parse` in a `try` and fallback to this `exec` behavior. I'm actually leaning toward just requiring chaining (`require('messageformat-loader!json-loader!./messages.json');`) to use this loader, though I could be swayed.